stdlib/public/core/CString.swift

```diff
@@ -162,39 +200,59 @@ extension String {
 
   @inlinable
   @_alwaysEmitIntoClient
-  public init?(validatingUTF8 cString: [CChar]) {
-    guard let length = cString.firstIndex(of: 0) else {
+  public init?(validatingCString nullTerminatedUTF8: [CChar]) {
```

Comment:
Can we more or less duplicate the docs for the other `validatingCString` initializer here?

Reply:
Good thought. These are stop-gap api that should be removed once pointer conversions can be restricted, but the time horizon is so long that it makes sense to document them until then. I should probably have done that when adding them last year.

Reply:
I will do the documentation update you suggest in a separate PR.

Reply:
Addressing the documentation issues in https://github.com/apple/swift/pull/71224